tests/raw_query/tests.py

```diff
@@ -274,3 +274,16 @@ def test_subquery_in_raw_sql(self):
             list(Book.objects.raw('SELECT id FROM (SELECT * FROM raw_query_book WHERE paperback IS NOT NULL) sq'))
         except InvalidQuery:
             self.fail("Using a subquery in a RawQuerySet raised InvalidQuery")
+
+    def test_db_column_name_is_used_in_raw_query(self):
+        """
+        Regression test that ensures the `column` attribute on the field is
+        used to generate the list of fields included in the query, as opposed
+        to the `attname`. This is important when the primary key is a
+        ForeignKey field because `attname` and `column` are not necessarily the
+        same.
+        """
+        try:
+            list(BookInfo.objects.raw('SELECT book FROM raw_query_bookinfo'))
```

Comment:
If possible, it would be nice to verify the query results instead of using this try/except pattern (which hides the traceback and makes a test failure more difficult to debug). Not sure if there's a good reason for the previous test using the pattern. The only thing I can think of is cross-database compatibility.